### DISCUSSION TITLE
Unrecoverable realm

### DISCUSSION THREAD

**Discussion by 6Kryptic** [ppy/osu#19648] [Q&A] [answered]
### Type

Game behaviour

### Bug description

I opened osu lazer to see all my beatmaps and skins deleted including my local scores. Somehow my realm had gotten corrupted and this had happened to one of my friends too. I've updated the game to see if anything would change but nothing did I'm not sure what to do.

### Screenshots or videos

![image](https://user-images.githubusercontent.com/107933517/183433313-7f4e1b26-4928-414e-b068-678c9fe31b5a.png)

### Version

2022.723.0

### Logs

[performance-gamethread.draw.log](https://github.com/ppy/osu/files/9282214/performance-gamethread.draw.log)
[performance-gamethread.input.log](https://github.com/ppy/osu/files/9282215/performance-gamethread.input.log)
[performance-gamethread.update.log](https://github.com/ppy/osu/files/9282216/performance-gamethread.update.log)
[runtime.log](https://github.com/ppy/osu/files/9282217/runtime.log)
[updater.log](https://github.com/ppy/osu/files/9282218/updater.log)
[database.log](https://github.com/ppy/osu/files/9282219/database.log)
[network.log](https://github.com/ppy/osu/files/9282220/network.log)
[performance.log](https://github.com/ppy/osu/files/9282221/performance.log)
[performance-gamethread.audio.log](https://github.com/ppy/osu/files/9282222/performance-gamethread.audio.log)

**Comment by peppy** [comment, marked as answer]
Can you please share all the `.realm` files?

**Reply by 6Kryptic** [reply]
Where would I share this? I can't share it here.

**Reply by frenzibyte** [reply]
Compress them in an archive file and attach it here.

**Reply by 6Kryptic** [reply]
https://github.com/6Kryptic/REALM

**Reply by peppy** [reply]
To recover, please delete `client.realm` and rename the large 9.2mb "corrupt" realm (`client_1659965118_corrupt.realm`) to `client.realm`. It should work fine.

**Reply by 6Kryptic** [reply]
thank god it works thanks alot!!

**Reply by peppy** [reply]
Great to hear. If it happens again, please save a copy of your log files immediately (without restarting the game) so we can try and learn what happened.

**Reply by 6Kryptic** [reply]
Okay coolio. Thanks again!
